### PR TITLE
core&ui: add problems gql to fix autocomplete

### PR DESCRIPTION
**Pull request by pandadtdyy**

### CODE REVIEW

File: packages/hydrooj/src/handler/problem.ts [hunk 1]
```diff
@@ -80,6 +80,14 @@ registerResolver(
         return pdoc;
     },
 );
+registerResolver('Query', 'problems(ids: [Int])', '[Problem]', async (arg, ctx) => {
+    if (arg.ids?.length) {
+        const res = await problem.getList(ctx.args.domainId, arg.ids);
+        console.log(res);
```
Comment (undefined-moe) [inline packages/hydrooj/src/handler/problem.ts]: ?

File: packages/hydrooj/src/handler/problem.ts [hunk 2]
```diff
@@ -80,6 +80,14 @@ registerResolver(
         return pdoc;
     },
 );
+registerResolver('Query', 'problems(ids: [Int])', '[Problem]', async (arg, ctx) => {
+    if (arg.ids?.length) {
+        const res = await problem.getList(ctx.args.domainId, arg.ids);
```
Comment (undefined-moe) [inline packages/hydrooj/src/handler/problem.ts]: use the arg indexByIdOnly

**Review by undefined-moe — CHANGES_REQUESTED** [review]

File: packages/hydrooj/src/handler/problem.ts (hunk 1 above)
Comment (pandadtdyy) [reply packages/hydrooj/src/handler/problem.ts]: 原本写了prefix 后来发现prefix可以直接用search

File: packages/hydrooj/src/handler/problem.ts [hunk 3]
```diff
@@ -80,6 +80,10 @@ registerResolver(
         return pdoc;
     },
 );
+registerResolver('Query', 'problems(ids: [Int])', '[Problem]', async (arg, ctx) => {
+    const res = await problem.getList(ctx.args.domainId, arg.ids, undefined, undefined, undefined, undefined, true);
+    return Object.keys(res).map((id) => res[+id]);
```
Comment (undefined-moe) [inline packages/hydrooj/src/handler/problem.ts:86]: is permission control required?
e.g. query hidden problem (in contest_edit) for admin user or problem creator

**Review by undefined-moe — CHANGES_REQUESTED** [review]

**Review by undefined-moe — APPROVED** [review]

**Pull request merged**